### PR TITLE
executor: make batch size of index join increasing slowly.

### PR DESCRIPTION
**Pull request by hanfei1991**

### CODE REVIEW

**Comment by zimulala** [comment]
LGTM

File: executor/index_lookup_join.go [hunk 1]
```diff
@@ -151,7 +153,10 @@ func (e *IndexLookUpJoin) Close() error {
 // Step6: fetch a batch of sorted "inner rows" based on the request rows.
 // Step7: do merge join on the **sorted** outer and inner rows.
 func (e *IndexLookUpJoin) Next() (Row, error) {
-	for ; e.resultCursor == len(e.resultBuffer); e.resultCursor = 0 {
+	for ; e.resultCursor == len(e.resultBuffer); e.resultCursor, e.curBatchSize = 0, e.curBatchSize*2 {
```
Comment (zz-jason) [inline executor/index_lookup_join.go]: 1. `s/e.curBatchSize = 0/e.resultCursor = 0`
2. consider overflow int

File: executor/index_lookup_join.go [hunk 2]
```diff
@@ -151,7 +153,10 @@ func (e *IndexLookUpJoin) Close() error {
...
+		if e.curBatchSize > e.batchSize {
```
Comment (zz-jason) [inline executor/index_lookup_join.go]: `curBatchSize` may overflow of max int

Comment (shenli) [reply executor/index_lookup_join.go]: Should we set a max value for curBatchSize?

Comment (hanfei1991) [reply executor/index_lookup_join.go]: I think we should set a max value for e.batchSize to make it less than max_int_64/2

Comment (zz-jason) [reply executor/index_lookup_join.go]: the max value of `e.batchSize` can be `e.batchSize`, `max_int_64/2` is too large

File: executor/index_lookup_join.go [hunk 3]
```diff
@@ -152,6 +154,9 @@ func (e *IndexLookUpJoin) Close() error {
 // Step7: do merge join on the **sorted** outer and inner rows.
 func (e *IndexLookUpJoin) Next() (Row, error) {
 	for ; e.resultCursor == len(e.resultBuffer); e.resultCursor = 0 {
+		if e.curBatchSize < e.batchSize {
```
Comment (shenli) [inline executor/index_lookup_join.go]: It is better to rename e.batchSize to e.maxBatchSize.

**Pull request merged**